### PR TITLE
Initial conductor implementation

### PR DESCRIPTION
**Pull request by jkeifer**

### CODE REVIEW

File: pkg/conductor/handler.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+	// force polling on start
+	h.NotifyNow()
+
+	conn, err := dbConf.Connect(ctx)
+	defer conn.Close(ctx)
+
+	if err != nil {
+		return err
+	}
+
```
Comment (jkeifer) [inline pkg/conductor/handler.go]: defer shouldn't be called until after the nil check

File: pkg/conductor/handler.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+		if err != nil {
+			return err
+		}
+
+		// TODO: need to handle backoff and retries, not just jump to failed
+		// TODO: we don't have a way to track attempts and therefore can't calculate backoff
+		return thread.InsertFailedEvent(ctx, conn, _err.Error())
+	}
+
+	err = handleFn()
```
Comment (jkeifer) [inline pkg/conductor/handler.go:67]: TODO comment about handling errors

File: pkg/conductor/argo.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+	return &ArgoClient{
+		config:        ac,
+		ctx:           ctx,
+		apiClient:     client,
+		serviceClient: client.NewWorkflowServiceClient(),
+		namespace:     namespace,
+		workflows:     workflows,
+	}, nil
+}
```
Comment (jkeifer) [inline pkg/conductor/argo.go:114]: TODO comment: we should add a check for template method to the workflow struct, then iterate through all our workflows here to validate that we have templates defined in the cluster for each of them -- allows us to fail fast if a template is missing.

File: cmd/conductor.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			Args:  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+			Run: func(cmd *cobra.Command, args []string) {
+				sc, err := conf.Parse()
+				if err != nil {
+					log.Fatal(err)
+				}
+				err = cmdutil.Run(
+					"swoop-conductor",
+					&conductor.PgConductor{
+						InstanceName: args[0],
```
Comment (jkeifer) [inline cmd/conductor.go]: We should make this a required option, not an arg, and then it can be specified via an env var.

**Review by hectormachin — APPROVED** [review]

**Pull request merged**